src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/GenerateCoverSheetService.java

```diff
@@ -54,38 +56,43 @@ public CaseDocument generateApplicantCoverSheet(final CaseDetails caseDetails, f
-        Object solicitorAddress = caseDetails.getData().get("solicitorAddress");
+        Map applicantAddress = (Map) caseDetails.getData().get("applicantAddress");
+        Map solicitorAddress = (Map) caseDetails.getData().get("solicitorAddress");
 
-        if (solicitorAddress != null) {
-            caseDetails.getData().put(BULK_PRINT_COVER_SHEET, getBulkPrintCoverSheet(bulkPrintCoverSheetBuilder, (Map) solicitorAddress));
-        } else if (respondentAddress != null) {
-            caseDetails.getData().put(BULK_PRINT_COVER_SHEET,  getBulkPrintCoverSheet(bulkPrintCoverSheetBuilder, (Map) respondentAddress));
+        if (addressLineOnePostCodeIsNotEmpty(solicitorAddress)) {
+            caseDetails.getData().put(BULK_PRINT_COVER_SHEET, getBulkPrintCoverSheet(bulkPrintCoverSheetBuilder, solicitorAddress));
+        } else if (addressLineOnePostCodeIsNotEmpty(applicantAddress)) {
+            caseDetails.getData().put(BULK_PRINT_COVER_SHEET,  getBulkPrintCoverSheet(bulkPrintCoverSheetBuilder, applicantAddress));
         }
     }
 
-    private void prepareRespondentCoverSheet(CaseDetails caseDetails) {
+    void prepareRespondentCoverSheet(CaseDetails caseDetails) {
         BulkPrintCoverSheet.BulkPrintCoverSheetBuilder bulkPrintCoverSheetBuilder = BulkPrintCoverSheet.builder()
             .ccdNumber(caseDetails.getId().toString())
             .recipientName(join(nullToEmpty(caseDetails.getData().get("appRespondentFMName")), " ",
                     nullToEmpty(caseDetails.getData().get("appRespondentLName"))));
 
-        Object respondentAddress = caseDetails.getData().get("respondentAddress");
-        Object solicitorAddress = caseDetails.getData().get("rSolicitorAddress");
+        Map respondentAddress = (Map) caseDetails.getData().get("respondentAddress");
+        Map solicitorAddress = (Map) caseDetails.getData().get("rSolicitorAddress");
 
-        if (solicitorAddress != null) {
-            caseDetails.getData().put(BULK_PRINT_COVER_SHEET, getBulkPrintCoverSheet(bulkPrintCoverSheetBuilder, (Map) solicitorAddress));
-        } else if (respondentAddress != null) {
-            caseDetails.getData().put(BULK_PRINT_COVER_SHEET,  getBulkPrintCoverSheet(bulkPrintCoverSheetBuilder, (Map) respondentAddress));
+        if (addressLineOnePostCodeIsNotEmpty(solicitorAddress)) {
+            caseDetails.getData().put(BULK_PRINT_COVER_SHEET, getBulkPrintCoverSheet(bulkPrintCoverSheetBuilder, solicitorAddress));
+        } else if (addressLineOnePostCodeIsNotEmpty(respondentAddress)) {
+            caseDetails.getData().put(BULK_PRINT_COVER_SHEET,  getBulkPrintCoverSheet(bulkPrintCoverSheetBuilder, respondentAddress));
         }
     }
 
+    private boolean addressLineOnePostCodeIsNotEmpty(Map address) {
+        return  ObjectUtils.isNotEmpty(address) && StringUtils.isNotEmpty((String) address.get("AddressLine1"))
+                && StringUtils.isNotEmpty((String) address.get("PostCode"));
```

Comment:
if you just want to check "not empty" a more clear solution will be to use `ObjectUtils.isNotEmpty` which will also cater for strings - no casting necessary. Unless you care about blank strings, then use `StringUtils.isNotBlank`

Reply:
Change made